#!/usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE = "sphero_formation"

gen = ParameterGenerator()

gen.add("alignment_factor", double_t, 0, "Alignment factor", 1.5, 0, 2)
gen.add("cohesion_factor", double_t, 0, "Cohesion factor", 1, 0, 2)
gen.add("separation_factor", double_t, 0, "Separation factor", 1, 0, 2)
gen.add("avoid_factor", double_t, 0, "Avoid factor", 1, 0, 2)
gen.add("max_speed", double_t, 0, "Maximum speed", 0.25, 0, 2)
gen.add("max_force", double_t, 0, "Maximum force", 0.18, 0, 2)
gen.add("friction", double_t, 0, "Friction", 0, 0, 0.2)
gen.add("crowd_radius", double_t, 0, "Crowding avoid radius", 0.45, 0, 2)
gen.add("search_radius", double_t, 0, "Neighbour searching radius", 0.75, 0, 3)
gen.add("avoid_radius", double_t, 0, "Radius for obstacle avoidance", 0.5, 0, 3)


exit(gen.generate(PACKAGE, "dynamic_reconfigure_node", "Reynolds"))
